### ISSUE TITLE
Reader View available for vimeo.com pages, but fails to display them

### ISSUE THREAD

**Issue by avaida** [mozilla/readability#134]
**Affected branches**:
Nightly 40.0a1 (2015-04-09), Aurora 39.0a2 (2015-04-09)

**Error message displayed**:
Failed to load article from page

**Sample pages showing this issue**:
* https://vimeo.com/channels/staffpicks/122258599
* https://vimeo.com/channels/staffpicks/124073656

**Notes**:
Reader View is not available for these pages on Beta 38.0b2-build1 (20150406174117).

**Comment by n1k0** [comment]
Woops.

**Comment by n1k0** [comment]
Do we actually want to support Vimeo video pages in reader mode? Put another way, do we want to detect them as possibly readable?

FWIW, Safari doesn't.

**Comment by gijsk** [comment]
I think we shouldn't, no...

**Comment by pdehaan** [comment]
Why wouldn't Vimeo be readable?
Why can't you show me an inline video and description?
Same w/ YouTubes: https://www.youtube.com/watch?v=VSYjs5RnlRo.

I believe that embed.ly supports both [Vimeo](http://embed.ly/docs/explore/extract?url=https%3A%2F%2Fvimeo.com%2Fchannels%2Fstaffpicks%2F122258599) and [YouTube](http://embed.ly/docs/explore/extract?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DVSYjs5RnlRo) and a bunch of other [providers](http://embed.ly/providers). Not to say that this needs to be a V1 or V2, thing, but I think it'd be nice to support videos and a larger set of sites if possible.

**Comment by n1k0** [comment]
> Why wouldn't Vimeo be readable?

In this case, *Viewable* would be a better word :) Having investigated this case for a little while, I can say that's it's much harder than it seems, especially considering the current code architecture. Many significant changes would be required to support it, with high risk of breaking what's currently supported. We're 4 weeks ahead only from release…

Using embed.ly could be a quick solution here, though:

- That would mean consuming an external REST API, so we'd depend on network connectivity; been told in the past that's not something possible/wanted atm in our build process;
- I'm not sure we actually want/have the time to evaluate their code (but maybe I'm wrong?)
- That would mean adding site-specific rules for these video sites, which is a strategy we recently discarded (though I suspect this could be reevaluated seeing so many issues where it could solve problems more easily than trying to be generic)…

I agree that in theory we should try supporting as many pages as we can, though for this case it's probably something to target for a post-38.1 version.

**Comment by gijsk** [comment]
Agreed with @n1k0 . I think for now we should try to figure out why we show the button and if we can avoid doing that. @n1k0, ideas?

**Comment by gijsk** [comment]
Both examples mentioned now produce something (rather than nothing/errors) in reader mode.
